pkg/scalers/scalersconfig/typed_config_test.go

```diff
@@ -0,0 +1,482 @@
+/*
+Copyright 2024 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scalersconfig
+
+import (
+	"net/url"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+// TestBasicTypedConfig tests the basic types for typed config
+func TestBasicTypedConfig(t *testing.T) {
+	RegisterTestingT(t)
+
+	sc := &ScalerConfig{
+		TriggerMetadata: map[string]string{
+			"stringVal":       "value1",
+			"intVal":          "1",
+			"boolValFromEnv":  "boolVal",
+			"floatValFromEnv": "floatVal",
+		},
+		ResolvedEnv: map[string]string{
+			"boolVal":  "true",
+			"floatVal": "1.1",
+		},
+		AuthParams: map[string]string{
+			"auth": "authValue",
+		},
+	}
+
+	type testStruct struct {
+		StringVal string  `keda:"name=stringVal, parsingOrder=triggerMetadata"`
```

Comment:
Its related to the above comment I have, can we have a test case where we don't specify `parsingOrder` at all?

Reply:
sure thing, I will add a case for a couple of combinations of empty `parsingOrder` along with `default` and `optional`.